src/Payone/Webhook/MessageBus/MessageHandler/NotificationForwardHandler.php

```diff
@@ -4,152 +4,84 @@
+use Symfony\Component\Messenger\Envelope;
+use Symfony\Component\Messenger\MessageBusInterface;
+use Symfony\Component\Messenger\Stamp\DelayStamp;
 
-class NotificationForwardHandler implements MessageSubscriberInterface
+#[AsMessageHandler(handles: NotificationForwardMessage::class)]
+class NotificationForwardHandler
 {
     public function __construct(
+        private readonly EntityRepository $forwardTargetRepository,
         private readonly EntityRepository $notificationForwardRepository,
-        private readonly Logger $logger
+        private readonly LoggerInterface $logger,
+        private readonly MessageBusInterface $messageBus
     ) {
     }
 
-    public function __invoke(NotificationForwardCommand $message): void
+    public function __invoke(NotificationForwardMessage $message): void
     {
-        $this->handle($message);
-    }
-
-    public function handle(NotificationForwardCommand $message): void
-    {
-        $notificationForwards = $this->getNotificationForwards($message->getNotificationTargetIds(), $message->getContext());
-        $multiHandle = curl_multi_init();
-
-        $this->logger->info('Forwarding notifications', array_keys($notificationForwards->getElements()));
-
-        $forwardRequests = $this->getForwardRequests($multiHandle, $notificationForwards);
-
-        if (empty($forwardRequests)) {
-            curl_multi_close($multiHandle);
-
+        $target = $this->forwardTargetRepository->search(new Criteria([$message->getNotificationTargetId()]), Context::createDefaultContext())->first();
+        if (!$target instanceof PayonePaymentNotificationTargetEntity) {
+            // should never occur - just to be safe.
             return;
         }
 
-        do {
-            $status = curl_multi_exec($multiHandle, $active);
-
-            if ($active) {
-                curl_multi_select($multiHandle);
+        $ch = curl_init();
+        curl_setopt($ch, \CURLOPT_URL, $target->getUrl());
+        curl_setopt($ch, \CURLOPT_HEADER, false);
+        curl_setopt($ch, \CURLOPT_POST, true);
+        curl_setopt($ch, \CURLOPT_RETURNTRANSFER, true);
+        curl_setopt($ch, \CURLOPT_TIMEOUT, 10);
+        curl_setopt($ch, \CURLOPT_CONNECTTIMEOUT, 10);
+        curl_setopt($ch, \CURLOPT_POSTFIELDS, http_build_query($message->getRequestData()));
+        curl_setopt($ch, \CURLOPT_FAILONERROR, true);
+        curl_setopt($ch, \CURLOPT_HTTPHEADER, $this->buildHeaders($message, $target));
+
+        $responseContent = (string)curl_exec($ch);
+        $responseInfo = curl_getinfo($ch);
+        $statusCode = curl_getinfo($ch, \CURLINFO_RESPONSE_CODE);
+        curl_close($ch);
+
+        $this->statusLogger($responseInfo, $responseContent, $message);
+        $this->saveNotificationForward($responseContent, $message);
+
+        if ($statusCode < 200 || $statusCode >= 300) {
+            $minuteWaitList = [2 => 5, 3 => 30, 4 => 120]; // maybe we will make this configurable in the future
```

Comment:
Can you at least move it to a constant? This could be called "ATTEMPT_WAIT_TIME_MAPPING", for example.

Reply:
done.